src/status_im/communities/core.cljs

```diff
@@ -780,3 +780,23 @@
                                 :community-id community-id
                                 :public-key   public-key
                                 :role-id      role-id})]})
+
+(rf/defn mute-chat-failed
+  {:events [:community/mute-community-failed]}
+  [{:keys [db]} community-id muted? error]
+  (log/error "mute community failed" community-id error)
+  {:db (assoc-in db [:communities community-id :muted] (not muted?))})
+
+(rf/defn mute-community-successfully
+  {:events [:community/mute-community-successful]}
+  [_ community-id]
+  (log/debug "muted community successfully" community-id))
+
+(rf/defn set-community-muted
+  {:events [:community/set-muted]}
+  [{:keys [db]} community-id muted?]
+  {:db (assoc-in db [:communities community-id :muted] muted?)
+   :json-rpc/call [{:method     "wakuext_setCommunityMuted"
+                    :params     [community-id muted?]
+                    :on-error   #(log/error "mute community failed" %)
```

Comment:
@jo-mut, @J-Son89, this is one example of an optimistic update that does not rollback the state if there was a failure. I think we should fix this because it generally means the end-user might take further actions on the app based on an incorrect view/state of the world.

Here the correct approach is to dispatch a separate event `:community/set-muted-error` and rollback the state.